src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -687,7 +687,13 @@ public static bool IsEnumType(this ITypeSymbol type, [NotNullWhen(true)] out INa
             var hasPrivateField = false;
             foreach (var member in type.GetMembers())
             {
-                if (!(member is IFieldSymbol fieldSymbol))
+                // TODO: Events?
+#if CODE_STYLE
+                var fieldSymbol = member as IFieldSymbol;
+#else
+                var fieldSymbol = member as IFieldSymbol ?? (member as IPropertySymbol)?.AssociatedField;
+#endif
```

Comment:
With this change, the code style file will be broken when running against newer versions of Roslyn that do not have property backing fields in GetMembers.

Reply:
In fact, this is causing at least some of the test failures.